tests/behat/src/FixtureContext.php

```diff
@@ -0,0 +1,281 @@
+
+    /**
+     * @Given /^I edit the campaign "([^"]*)"$/
+     * @param $name
+     */
+    public function iEditTheCampaign($name)
+    {
+        $item = $this->getCampaign($name);
+        assertNotNull($item, sprintf('Campaign %s not found', $name));
+
+        $button = $item->find('css', '.font-icon-cog');
+        assertNotNull($button, sprintf('Campaign %s has no edit button', $name));
+        $button->click();
+    }
+
+    /**
+     * @Given /^I delete the campaign "([^"]*)"$/
+     * @param $name
+     */
+    public function iDeleteTheCampaign($name)
+    {
+        $item = $this->getCampaign($name);
+        assertNotNull($item, sprintf('Campaign %s not found', $name));
+
+        $button = $item->find('css', '.font-icon-cancel');
+        assertNotNull($button, sprintf('Campaign %s has no delete button', $name));
+        $button->click();
+
+        /*@ var Behat\Mink\Driver\Selenium2Driver $driver */
+        $driver = $this->getMainContext()->getSession()->getDriver();
+        $driver->getWebDriverSession()->accept_alert();
+    }
+
+    /**
+     * @Then /^I should see the "([^"]*)" form$/
+     * @param string $id HTML ID of form
+     */
+    public function iShouldSeeTheForm($id)
+    {
+        /** @var DocumentElement $page */
+        $page = $this->getMainContext()->getSession()->getPage();
+        $form = $page->find('css', "form#{$id}");
+        assertNotNull($form, "form with id $id could not be found");
+        assertTrue($form->isVisible(), "form with id $id is not visible");
+    }
+
+    /**
+     * Checks that the message box contains specified text.
+     *
+     * @Then /^I should see "(?P<text>(?:[^"]|\\")*)" in the message box$/
+     * @param string $text
+     */
+    public function assertMessageBoxContainsText($text)
+    {
+        /** @var FeatureContext $mainContext */
+        $mainContext = $this->getMainContext();
+        $mainContext
+            ->assertSession()
+            ->elementTextContains('css', '.message-box', str_replace('\\"', '"', $text));
+    }
+
+    /**
+     * @Then I should see a modal titled :title
+     * @param string $title
+     */
+    public function iShouldSeeAModalTitled($title)
+    {
+        $page = $this->getMainContext()->getSession()->getPage();
+        $modalTitle = $page->find('css', '[role=dialog] .modal-header > .modal-title');
+        assertNotNull($modalTitle, 'No modal on the page');
+        assertTrue($modalTitle->getText() == $title);
+    }
+
+    /**
+     * @Then /^I close the modal$/
+     */
+    public function iCloseTheModal()
+    {
+        /** @var DocumentElement $page */
+        $page = $this->getMainContext()->getSession()->getPage();
+        $button = $page->find('css', '.modal-header .close');
+        assertNotNull($button, 'Close button not found');
+
+        $button->click();
+    }
+
+    /**
+     * @Given /^I click the "([^"]+)" gallery item$/
+     * @param $name
+     */
+    public function iClickTheGalleryItem($name)
```

Comment:
Add the AssetAdmin version of FixtureContext to your behat.yml as well, and use stepISelectGalleryItem() instead of duplicating.

Reply:
Same as above. Can't use two descendants of the base `FixtureContext`

Reply:
Make your new one extend the prior instead?

Reply:
We decided to go with the inline copy for now; It's not that big a cost to have a minor overlap.  :)